cluster_orchestrator/cluster-scheduler/calculation.py

```diff
@@ -83,7 +83,7 @@ def deploy_on_best_among_desired_nodes(job: dict, nodes) -> Union[dict, Negative
     else:
         desired_nodes_list = nodes.split(";")
         for node in active_nodes:
-            if node["node_info"]["host"] in desired_nodes_list:
+            if node.get("node_info").get("host") in desired_nodes_list:
```

Comment:
if node_info is not available this still returns None right?

What about something like this?
```suggestion
            if node.get("node_info",{}).get("host") in desired_nodes_list:
```

Reply:
Good catch! Thanks